internal/otelcollector/config/otlpexporter/env_vars.go

```diff
@@ -78,3 +104,11 @@ func makeOtlpEndpointVariable(pipelineName string) string {
 func makeBasicAuthHeaderVariable(pipelineName string) string {
 	return fmt.Sprintf("%s_%s", basicAuthHeaderVariablePrefix, envvar.MakeEnvVarCompliant(pipelineName))
 }
+
+func makeHeaderVariable(header telemetryv1alpha1.Header, pipelineName string) string {
+	return fmt.Sprintf("HEADER_%s_%s", envvar.MakeEnvVarCompliant(pipelineName), envvar.MakeEnvVarCompliant(header.Name))
+}
+
+func makeTLSVariable(prefix, pipelineName string) string {
```

Comment:
This function is more generic than it's name. Let's either make 3 specific function for each purpose or use a more generic name.